### ISSUE TITLE
[@wry/equality]: equality fails check on these values

### ISSUE THREAD

**Issue by ScottAwesome** [benjamn/wryware#150]
Thanks for maintaing such a great library!

I was recently trying to debug a program, and I thought some equality checks were an issue, I use `@wry/equality` for this. When I ran it through my own test (to see if it would fail on equality check), it in fact, did:

```js
exports.foo = {
  foo: 'value1',
  bar: new Set([1, 2, 3]),
  baz: /foo/i,
  bat: {
    hello: new Map([ ['hello', 'world'] ]),
    world: {
      aaa: new Map([
        [{ foo: /bar/ }, 'sub sub value1'],
      ]),
      bbb: [1, 2, { prop2:1, prop:2 }, 4, 5]
    }
  },
  quz: new Set([{ a:1 , b:2 }]),
  qut: new Date(2016, 2, 10),
  qar: new Uint8Array([1, 2, 3, 4, 5]),
}

exports.bar = {
  quz: new Set([{ a:1 , b:2 }]),
  baz: /foo/i,
  foo: 'value1',
  bar: new Set([1, 2, 3]),
  qar: new Uint8Array([1, 2, 3, 4, 5]),
  qut: new Date('2016/03/10'),
  bat: {
    world: {
      aaa: new Map([
        [{ foo: /bar/ }, 'sub sub value1'],
      ]),
      bbb: [1, 2, { prop2:1, prop:2 }, 4, 5]
    },
    hello: new Map([ ['hello', 'world'] ])
  }
};
```
keys and values are the same, they are just in different order between the two objects.

Is this intended? As I imagine key order should _not_ play a part here no?

**Comment by benjamn** [comment]
Thanks for the (confirmed failing) test case @ScottAwesome!

I'm guessing this is because we have no special handling for `Uint8Array` objects (yet), so they're compared by reference rather than by structure.

If you have any time to submit a PR with your code as a failing test case, and/or tackle the `Uint8Array` case yourself (no obligation—only if that sounds like fun and you have time), I will hold off on doing those things myself. Otherwise, I can take a closer look at this in the next week or so.

**Comment by ScottAwesome** [comment]
@benjamn I opened a PR for this: https://github.com/benjamn/wryware/pull/155

**Comment by benjamn** [comment]
@ScottAwesome This should be fixed in `@wry/equality@0.5.0`, just published to npm. Thanks for the PR!
